### PR TITLE
Issue 317

### PR DESCRIPTION
**Pull request by bennahugo**
Make tests more robust

### CODE REVIEW

**Comment by bennahugo** [comment]
@SpheMakh Looks like the ngc test you put in never worked in the first place - fails in source finding. Trying to fix it now, but may revert it to the initial test I had

**Comment by bennahugo** [comment]
It looks like it is working

**Comment by bennahugo** [comment]
You've grossly undersampled the input images. So I just changed it.

File: stimela/tests/acceptance_tests/stimela-test-ngc417.py [hunk 1]
```diff
@@ -563,9 +563,9 @@ def testEndToEndReduction(self):
                         "msname"        :   MS,
                         "channelrange"  :   chans,               #Other channels don't have any data   
                         "weight"        :   "briggs 0",               # Use Briggs weighting to weigh visibilities for imaging
-                        "npix"          :   1026,                   # Image size in pixels
-                        "trim"          :   1024,                    # To avoid aliasing
-                        "cellsize"      :   2.5,                      # Size of each square pixel
+                        "npix"          :   8196,                   # Image size in pixels
+                        "trim"          :   5000,                    # To avoid aliasing
```
Comment (SpheMakh) [inline stimela/tests/acceptance_tests/stimela-test-ngc417.py]: trim has been deprecated in wsclean. Better to use -padding

File: stimela/tests/acceptance_tests/stimela-test-ngc417.py [hunk 2]
```diff
@@ -563,9 +563,9 @@ def testEndToEndReduction(self):
                         "msname"        :   MS,
                         "channelrange"  :   chans,               #Other channels don't have any data   
                         "weight"        :   "briggs 0",               # Use Briggs weighting to weigh visibilities for imaging
-                        "npix"          :   1026,                   # Image size in pixels
-                        "trim"          :   1024,                    # To avoid aliasing
-                        "cellsize"      :   2.5,                      # Size of each square pixel
+                        "npix"          :   8196,                   # Image size in pixels
```
Comment (SpheMakh) [inline stimela/tests/acceptance_tests/stimela-test-ngc417.py]: These images are too big for a test!

Comment (bennahugo) [reply stimela/tests/acceptance_tests/stimela-test-ngc417.py]: Lets first get it back to passing before worrying about image size

**Comment by bennahugo** [comment]
Changes added please update review @SpheMakh

**Comment by bennahugo** [comment]
retest this please

**Comment by bennahugo** [comment]
retest this please

**Comment by bennahugo** [comment]
@SpheMakh There is a problem with the rhodes gateway DNS 
```
Collecting pyyaml
[91m  Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7f6124a3dd90>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/pyyaml/
[0m[91m  Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7f6124a3d910>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/pyyaml/
[0m[91m  Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7f6124a3d290>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/pyyaml/
[0m[91m  Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7f6124a3df10>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/pyyaml/
[0m[91m  Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7f6124a3d590>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/pyyaml/
[0m[91m  Could not find a version that satisfies the requirement pyyaml (from versions: )
[0m[91mNo matching distribution found for pyyaml
[0mRemoving intermediate container a49cb80663df
```

**Comment by SpheMakh** [comment]
Should we force this PR through?

**Comment by bennahugo** [comment]
Well I don't know if the test works or not. I don't have enough memory to test it locally

**Comment by bennahugo** [comment]
I'm trying to play with the resolv config to see if we can switch to the google DNS server

**Comment by SpheMakh** [comment]
Ok, lets hold off then

**Comment by bennahugo** [comment]
retest this please

**Comment by bennahugo** [comment]
@SpheMakh I need to flush the iptables and restart the docker service to repopulate them. Unfortunately I have some DDFacet tests running at the moment that I don't want to interrupt, so this will only happen sometime later tommorrow

**Comment by bennahugo** [comment]
retest this please

**Comment by SpheMakh** [comment]
images still building on docker hub. Will re-run test later

**Comment by SpheMakh** [comment]
chgcentre and montages images failed to build

**Comment by SpheMakh** [comment]
retest this please

**Comment by SpheMakh** [comment]
This test has been building the casa_plotcal image for more than 12 hours. @bennahugo could you please kill the process

**Comment by bennahugo** [comment]
@SpheMakh no it seems to be running. Already started testing. In the singularity test right now, see
https://jenkins.meqtrees.net/job/stimela-pr/ws/254/test-output/log-vla_ngc417_reduction_script.txt

**Comment by bennahugo** [comment]
Never mind it looks like singularity is in a permanent sleep state. Maybe there should be a time limit switch after which the process is killed?

**Comment by bennahugo** [comment]
Going to cancel this - still hanging indefinitely
```
2019-02-06 10:01:38,256 - STIMELA - INFO - ---------------------------------
2019-02-06 10:01:38,257 - STIMELA - INFO - Stimela version 1.0.1
2019-02-06 10:01:38,258 - STIMELA - INFO - Sphesihle Makhathini <sphemakh@gmail.com>
2019-02-06 10:01:38,259 - STIMELA - INFO - Running: Singularity Test
2019-02-06 10:01:38,260 - STIMELA - INFO - ---------------------------------
2019-02-06 10:01:38,284 - STIMELA - INFO - Adding cab '/var/lib/jenkins/jobs/stimela-pr/workspace/255/singularity_images/stimela_simms_1.0.1.img' to recipe. The container will be named 'simms_example'
2019-02-06 10:01:38,300 - STIMELA - INFO - Adding cab '/var/lib/jenkins/jobs/stimela-pr/workspace/255/singularity_images/stimela_casa_1.0.1.img' to recipe. The container will be named 'listobs_example'
2019-02-06 10:01:38,337 - STIMELA - INFO - Adding cab '/var/lib/jenkins/jobs/stimela-pr/workspace/255/singularity_images/stimela_meqtrees_1.0.1.img' to recipe. The container will be named 'simulator_example'
2019-02-06 10:01:38,398 - STIMELA - INFO - Adding cab '/var/lib/jenkins/jobs/stimela-pr/workspace/255/singularity_images/stimela_casa_1.0.1.img' to recipe. The container will be named 'plot_vis'
2019-02-06 10:01:38,487 - STIMELA - INFO - Adding cab '/var/lib/jenkins/jobs/stimela-pr/workspace/255/singularity_images/stimela_wsclean_1.0.1.img' to recipe. The container will be named 'imager_example_robust_0'
2019-02-06 10:01:38,573 - STIMELA - INFO - Adding cab '/var/lib/jenkins/jobs/stimela-pr/workspace/255/singularity_images/stimela_wsclean_1.0.1.img' to recipe. The container will be named 'imager_example_robust_1'
2019-02-06 10:01:38,660 - STIMELA - INFO - Adding cab '/var/lib/jenkins/jobs/stimela-pr/workspace/255/singularity_images/stimela_wsclean_1.0.1.img' to recipe. The container will be named 'imager_example_robust_2'
2019-02-06 10:01:38,663 - STIMELA - INFO - Adding cab '/var/lib/jenkins/jobs/stimela-pr/workspace/255/singularity_images/stimela_casa_1.0.1.img' to recipe. The container will be named 'delete_ms'
2019-02-06 10:01:38,664 - STIMELA - INFO - Running job simms_example
2019-02-06 10:01:38,665 - STIMELA - INFO - STEP 1 :: Creating MS
2019-02-06 10:01:38,667 - STIMELA - INFO - Instantiating container [NAG]. The container ID is printed below.
```

**Comment by bennahugo** [comment]
```
2019-02-06 10:01:36,116 - STIMELA - INFO - STEP 1 :: Creating MS
2019-02-06 10:01:36,118 - STIMELA - INFO - Instantiating container [simms_example-140429915915344154944009495]. The container ID is printed below.
running: docker create --user 114:119 --cpus 2.500000 --memory 2gb  -v /var/lib/jenkins/jobs/stimela-pr/workspace/255/projects/pyenv/local/lib/python2.7/site-packages/stimela:/scratch/stimela:ro -v /var/lib/jenkins/jobs/stimela-pr/workspace/255/test-output/stimela_parameter_files:/configs:ro -v /var/lib/jenkins/jobs/stimela-pr/workspace/255/test-output/msdir:/home/jenkins/msdir:rw -v /var/lib/jenkins/jobs/stimela-pr/workspace/255/projects/Stimela/stimela/tests/acceptance_tests/input:/input:ro -v /var/lib/jenkins/jobs/stimela-pr/workspace/255/test-output/output:/home/jenkins/output:rw  -e CONFIG=/configs/simms_example-140429915915344154944009495.json -e MSDIR=/home/jenkins/msdir -e INPUT=/input -e HOME=/home/jenkins/output -e OUTPUT=/home/jenkins/output -e LOGFILE=/home/jenkins/output/log-simms_example.txt  --name simms_example-140429915915344154944009495 --shm-size 1gb jenkins_cab/simms 
9fda13b2bca9d72cdb21a9f4c0013054e0c34b74e765215501103158fbaff6e4
running: docker start -a simms_example-140429915915344154944009495
running: simms --name "/home/jenkins/msdir/meerkat_simulation_example.ms" --tel "kat-7" --type "casa" --coord-sys "itrf" --direction J2000,0deg,-30deg --synthesis 0.128 --dtime 10 --freq0 750MHz --dfreq 1MHz --nband 1 --nchan 1 --pol "XX XY YX YY" --feed "perfect X Y" --nolog
/bin/sh: 1: casa: not found
mv: cannot stat './tmpxos3RU/log-simms.txt': No such file or directory
2019/2/6 8:1:37 ##INFO: Simms >>: msname="/home/jenkins/msdir/meerkat_simulation_example.ms", label="None", tel="kat-7", pos="/usr/local/lib/python2.7/dist-packages/simms/observatories/kat-7.itrf.txt", pos_type="ascii", synthesis=0.128, scan_length=None, dtime="10.0", freq0=['750MHz'], dfreq=['1MHz'], nchan=[1], stokes="XX XY YX YY", setlimits=False, elevation_limit=0.000000, shadow_limit=0.000000, coords="itrf",lon_lat="None", noup=False, nbands=1, direction=['J2000,0deg,-30deg'], outdir=".",date="None",fromknown=False, feed="perfect X Y",scan_lag=0,auto_corr=False,optimise_start=False
ERROR: simms.py returns errr code 127. Having Trouble accessing the MS. Something went wrong while creating the MS, please check the logs.
If you believe this is due to a bug in simms, please notify me via https://github.com/SpheMakh/simms/issues/new
Traceback (most recent call last):
  File "/usr/local/bin/simms", line 5, in <module>
    main()
  File "/usr/local/lib/python2.7/dist-packages/simms/simms.py", line 386, in main
    optimise_start=args.optimise_start)
  File "/usr/local/lib/python2.7/dist-packages/simms/simms.py", line 230, in create_empty_ms
    raise CasapyError(message)
simms.simms.CasapyError: Having Trouble accessing the MS. Something went wrong while creating the MS, please check the logs.
If you believe this is due to a bug in simms, please notify me via https://github.com/SpheMakh/simms/issues/new
Traceback (most recent call last):
  File "/scratch/code/run.py", line 41, in <module>
    utils.xrun(cab['binary'], args+['--nolog']+positional)
  File "/scratch/stimela/utils/__init__.py", line 75, in xrun
    raise SystemError('%s: returns errr code %d'%(command, process.returncode))
SystemError: simms: returns errr code 1
```

**Comment by SpheMakh** [comment]
Closing this PR for now

**Comment by bennahugo** [comment]
@SpheMakh why? The current test infrastructure will only work with changes in this PR

**Comment by SpheMakh** [comment]
I will reopen it in a couple of hours. I just don't want to trigger a test with the next couple of changes

**Comment by bennahugo** [comment]
Sure

**Pull request closed without merging**